deepchecks/base/dataset.py

```diff
@@ -88,13 +89,20 @@ def __init__(
             label_name (Optional[Hashable]):
                 If `label` is given, then this name is used as the column name for the labels.
                 If `label` is none, then looks for this name in the data dataframe.
-            use_default_index (bool, default False):
-                Whether to use the dataframe index as the index column, for index related checks. can't be used
-                together with `index_name`
             index_name (Optional[Hashable]):
-                Name of the index column in the DataFrame. can't be used together with `use_default_index`
+                Name of the index column in the dataframe. If create_index_from_dataframe_index is True and index_name
```

Comment:
here is the place to look @shir22 @ItayGabbay

Reply:
In general this looks good.
A few things that I found not inuitive:
1. The wording "create index" - I think it may confuse the user - if they're passing the index column and telling you "that is the index", what exactly are you "creating"?
2. Somewhat similar but different for "date", what does it mean to "create date".
3. I think in general our approach of "we can have a single index column" and "we can have a single date column", and each are treated differently isn't very clear to the user (what we use each for, and why can there be only a single of each), but I think this is something we should iterate with feedback to improve the clarity and usability of this

Reply:
1. Maybe "set_Index.."?
2. same
3. Don't have an idea for an alternative simple solution to start from...

Reply:
we can change the whole concept of "date" to "datetime", you think it's better?